packages/marshal/src/helpers/copyRecord.js

```diff
@@ -25,31 +21,27 @@ export const CopyRecordHelper = harden({
 
   canBeValid: (candidate, check) => {
     const reject = !!check && (details => check(false, details));
-    const proto = getPrototypeOf(candidate);
-    if (proto !== objectPrototype && proto !== null) {
-      return reject && reject(X`Unexpected prototype for: ${candidate}`);
+    if (getPrototypeOf(candidate) !== objectPrototype) {
+      return (
+        (reject &&
+        reject(X`Records must inherit from Object.prototype: ${candidate}`))
+      );
     }
-    const descs = getOwnPropertyDescriptors(candidate);
-    const descKeys = ownKeys(descs);
 
-    for (const descKey of descKeys) {
-      if (typeof descKey !== 'string') {
-        // Pass by copy
-        return (
-          (reject &&
-          reject(
-            X`Records can only have string-named own properties: ${candidate}`,
-          ))
-        );
-      }
-      const desc = descs[descKey];
-      if (canBeMethod(desc.value)) {
-        return (
-          (reject &&
-          reject(
-            X`Records cannot contain non-far functions because they may be methods of an implicit Remotable: ${candidate}`,
-          ))
-        );
+    for (const key of ownKeys(candidate)) {
+      const valid =
+        (typeof key === 'string' ||
+          (!!reject &&
+            reject(
+              X`Records can only have string-named properties: ${candidate}`,
+            ))) &&
+        (!canBeMethod(candidate[key]) ||
```

Comment:
Before this PR, `canBeMethod` tests `desc.value`. This PR tests `candidate[key]`. Where does the new code ensure that `candidate` is frozen and has no own accessor properties? Were we previously being reflective unnecessarily? Isn't reflective examination the safer style anyway? Is a reflective check significantly slower on XS?